lms/djangoapps/certificates/tests/test_webview_views.py

```diff
@@ -795,7 +795,41 @@ def test_render_html_view_with_preview_mode_when_user_already_has_cert(self):
         self.assertIn('course_title_0', response.content)
         self.assertIn('Signatory_Title 0', response.content)
 
-    @freeze_time('2017-09-10 00:00:00Z')
+    @override_settings(FEATURES=FEATURES_WITH_CERTS_ENABLED)
+    @ddt.data(
+        (-2, True),
+        (-2, False),
+        (10, False)
+    )
+    @ddt.unpack
+    def test_html_view_certificate_availability_date_for_instructor_paced_courses(self, cert_avail_delta, self_paced):
+        """
+        test certificate web view should display the certificate availability date
+        as the issued date for instructor-paced courses
+        """
+        self.course.self_paced = self_paced
+        self.course.certificate_available_date = datetime.datetime.today() + datetime.timedelta(cert_avail_delta)
+        self.course.save()
```

Comment:
Isn't `self.course` from the modulestore?  If so, `.save` doesn't do what you would expect.  You would want to call `self.store.update_item(self.course, ..)` to actually update the course.

Reply:
I'll change the call, but what does `self.course.save()` actually do, then? I've used it in a bunch of tests over the last year to do what Greg was doing here.

Reply:
Ah, never mind- I guess I've been using `update_item` in conjunction with `save()` and I just forgot.

Reply:
Hmm.. perhaps `.save` does do what you expect - if it eventually calls the modulestore.  I had never used that path before, but it may be there: https://github.com/edx/XBlock/blob/0496f693352d962f5b7c2482b3fc410c599e3505/xblock/mixins.py#L247

In either case, one should need to call either `store.update_item` or `block.save` - both shouldn't be needed.  And, perhaps either works - although I'm confident of the former working.

Reply:
Interesting. The rest of this file calls both. Maybe that's just to be safe, but in any case, I'll go with the one that we know works.